### PR TITLE
feat(react-select): use react-select version 4.3.1

### PR DESCRIPTION
**Pull request by YossiSaadi**
https://monday.monday.com/boards/3532714909/pulses/6864028773

### Why
We now have `react-windowed-select`, which, for some reason, does not use `react-select` as peer dep, but as a dep.
So this causes two versions of `react-select` at our bundle, as we cannot resolve both our dependency and `react-windowed-select`'s version to same version.

**This should be reviewed with caution**

The only breaking behavior on `react-select` 4 in compare to 3 is the `isMulti` behavior.
It used to return `null` when no options were selected. now it returns always array (empty array in case no options were selected).
So I had, to not cause breaking changes, mimic the current behavior with the new `react-select`'s api for `isMulti`'s `onChange`.

### CODE REVIEW

File: packages/core/src/components/Dropdown/__tests__/Dropdown.test.js [hunk 1]
```diff
@@ -150,7 +150,8 @@ describe("Dropdown", () => {
 
       expect(onChange).toBeCalledTimes(2);
       expect(onChange).toHaveBeenLastCalledWith(null, {
-        action: "clear"
+        action: "clear",
+        removedValues: [mockOptions[0]]
```
Comment (YossiSaadi) [inline packages/core/src/components/Dropdown/__tests__/Dropdown.test.js:154]: `onChange` now returns this in addition for when removing values in `isMulti` state

File: packages/core/src/components/Dropdown/Dropdown.tsx [hunk 2]
```diff
@@ -289,7 +289,8 @@ const Dropdown: VibeComponent<DropdownComponentProps, HTMLElement> & {
     );
     const onChange = (option: DropdownOption | DropdownOption[], meta: ActionMeta<DropdownOption>) => {
       if (customOnChange) {
-        customOnChange(option, meta);
+        const newValue = multi ? (option.length > 0 ? option : null) : option;
```
Comment (YossiSaadi) [inline packages/core/src/components/Dropdown/Dropdown.tsx:292]: if multi and array is empty return null, to mimic current behavior, and ignore the behavior of the new react-select's api for isMulti

File: packages/core/src/components/Dropdown/Dropdown.tsx [hunk 3]
```diff
@@ -260,7 +260,7 @@ const Dropdown: VibeComponent<DropdownComponentProps, HTMLElement> & {
           : selectedOptions;
 
         if (customOnChange) {
-          customOnChange(newSelectedOptions, e);
+          customOnChange(newSelectedOptions.length > 0 ? newSelectedOptions : null, e);
```
Comment (YossiSaadi) [inline packages/core/src/components/Dropdown/Dropdown.tsx:263]: I'm not 100% this is needed here
wdyt?

File: packages/core/src/components/Dropdown/__tests__/__snapshots__/Dropdown.snapshot.test.js.snap [hunk 4]
```diff
@@ -7,18 +7,18 @@ exports[`Dropdown renders correctly snapshot driver should open menu on click if
     id="dropdown-menu-id"
   >
     <span
+      aria-atomic="false"
       aria-live="polite"
-      class="css-1laao21-a11yText"
+      aria-relevant="additions text"
+      class="css-1f43avz-a11yText-A11yText"
     >
```
Comment (YossiSaadi) [inline packages/core/src/components/Dropdown/__tests__/__snapshots__/Dropdown.snapshot.test.js.snap:14]: all this new aria-live behavior is intended by react-select 4.0.0, does not affect us, only adds a11y

File: packages/core/src/components/Dropdown/__tests__/__snapshots__/Dropdown.snapshot.test.js.snap [hunk 5]
```diff
@@ -193,18 +193,18 @@ exports[`Dropdown renders correctly snapshot driver should open menu on focus if
...
-      >
-         
-      </span>
+        id="aria-selection"
+      />
       <span
         id="aria-context"
       >
-           0 results available.   Selected:  is focused ,type to refine list, press Down to open the menu, 
+            Selected:  is focused ,type to refine list, press Down to open the menu,
```
Comment (YossiSaadi) [inline packages/core/src/components/Dropdown/__tests__/__snapshots__/Dropdown.snapshot.test.js.snap:207]: It seem like it was a bug at react select. no reason that 0 results available would be read to user here, as the input does not filter the results at all. it should only appear if the input filter does not match any option. in here all the options are available

File: packages/core/src/components/Dropdown/__tests__/__snapshots__/Dropdown.snapshot.test.js.snap [hunk 6]
```diff
@@ -761,18 +797,18 @@ exports[`Dropdown renders correctly snapshot driver should use async if set 1`]
...
```
Comment (YossiSaadi) [inline packages/core/src/components/Dropdown/__tests__/__snapshots__/Dropdown.snapshot.test.js.snap:811]: Same. seem like it was a bug. please verify me

File: packages/core/src/components/Dropdown/__tests__/__snapshots__/Dropdown.snapshot.test.js.snap [hunk 7]
```diff
@@ -820,13 +856,13 @@ exports[`Dropdown renders correctly snapshot driver should use async if set 1`]
...
-              class="css-imwzes-Component"
+              class="css-1yvy2vo-LoadingDot"
             />
             <span
-              class="css-1ue98e-Component"
+              class="css-zoievk-LoadingDot"
             />
             <span
-              class="css-fwsb3e-Component"
+              class="css-x748d8-LoadingDot"
```
Comment (YossiSaadi) [inline packages/core/src/components/Dropdown/__tests__/__snapshots__/Dropdown.snapshot.test.js.snap:865]: module name changed in react-select

File: packages/core/src/components/Dropdown/__tests__/__snapshots__/Dropdown.snapshot.test.js.snap [hunk 8]
```diff
@@ -957,7 +993,7 @@ exports[`Dropdown renders correctly snapshot driver should use virtualization if
     >
       <div
         class=""
-        style="position: relative; height: 300px; overflow: auto; will-change: transform; direction: ltr; overflow-y: auto; box-sizing: border-box;"
+        style="position: relative; height: 300px; width: 100%; overflow: auto; will-change: transform; direction: ltr; overflow-y: auto; box-sizing: border-box;"
```
Comment (YossiSaadi) [inline packages/core/src/components/Dropdown/__tests__/__snapshots__/Dropdown.snapshot.test.js.snap:996]: not sure why they've added it, but it doesn't affect the ui it seems

File: packages/core/src/components/Dropdown/__tests__/__snapshots__/Dropdown.snapshot.test.js.snap [hunk 9]
```diff
@@ -1329,7 +1377,7 @@ exports[`Dropdown renders correctly when readOnly 1`] = `
         <input
           aria-autocomplete="list"
           aria-label="Readonly   Selected: "
-          className="css-62g3xt-dummyInput"
+          className="css-wmatm6-dummyInput-DummyInput"
```
Comment (YossiSaadi) [inline packages/core/src/components/Dropdown/__tests__/__snapshots__/Dropdown.snapshot.test.js.snap:1380]: module name changed (?)

**Comment by YossiSaadi** [comment]
@shaharzil @rivka-ungar @talkor need your cooperation on that one to verify me
the tests were not comprehensive enough to put 100% trust on it

File: packages/core/src/components/Dropdown/Dropdown.tsx (hunk 3 above)
Comment (talkor) [reply packages/core/src/components/Dropdown/Dropdown.tsx:263]: Can you elaborate on the change?

Comment (YossiSaadi) [reply packages/core/src/components/Dropdown/Dropdown.tsx:263]: I'm not sure newSelectedOptions could have been null at any scenario.
And if it cannot be null, it means our API is not consistent today, and can sometimes return null and sometimes empty array (?).

Comment (YossiSaadi) [reply packages/core/src/components/Dropdown/Dropdown.tsx:263]: So this change might be redundant? Or am I misunderstanding something?

Comment (talkor) [reply packages/core/src/components/Dropdown/Dropdown.tsx:263]: If it's not mandatory for the migration I wouldn't make changes here

File: packages/core/src/components/Dropdown/__tests__/__snapshots__/Dropdown.snapshot.test.js.snap (hunk 5 above)
Comment (talkor) [reply packages/core/src/components/Dropdown/__tests__/__snapshots__/Dropdown.snapshot.test.js.snap:207]: It's not breaking, right?

Comment (YossiSaadi) [reply packages/core/src/components/Dropdown/__tests__/__snapshots__/Dropdown.snapshot.test.js.snap:207]: it is only for the screen reader, and it is a bug fix, so don't think so

**Pull request merged**